src/spikeinterface/core/base.py

```diff
@@ -846,43 +846,63 @@ def save_to_memory(self, sharedmem=True, **save_kwargs) -> "BaseExtractor":
         return cached
 
     # TODO rename to saveto_binary_folder
-    def save_to_folder(self, name=None, folder=None, overwrite=False, verbose=True, **save_kwargs):
+    def save_to_folder(
+        self,
+        name: str = None,
+        folder: str | Path = None,
+        overwrite: str = False,
+        verbose: bool = True,
+        **save_kwargs,
+    ):
         """
-        Save extractor to folder.
+        Save the extractor and its data to a folder.
 
-        The save consist of:
-          * extracting traces by calling get_trace() method in chunks
-          * saving data into file (memmap with BinaryRecordingExtractor)
-          * dumping to json/pickle the original extractor for provenance
-          * dumping to json/pickle the cached extractor (memmap with BinaryRecordingExtractor)
+        This method extracts trace data, saves it to a file (using a memory-mapped format
+        with BinaryRecordingExtractor), and stores both the original extractor's provenance
+        and the cached extractor's metadata in JSON format.
 
-        This replaces the use of the old CacheRecordingExtractor and CacheSortingExtractor.
+        The folder final location and name can be specified in a couple of ways ways:
 
-        There are 2 option for the "folder" argument:
-          * explicit folder: `extractor.save(folder="/path-for-saving/")`
-          * explicit sub-folder, implicit base-folder : `extractor.save(name="extarctor_name")`
-          * generated: `extractor.save()`
+        1. Explicitly providing the full path:
+        ```
+        extractor.save_to_folder(folder="/path/to/save/")
+        ```
 
-        The second option saves to subfolder "extractor_name" in
-        "get_global_tmp_folder()". You can set the global tmp folder with:
-        "set_global_tmp_folder("path-to-global-folder")"
+        2. Providing a subfolder name, with the base folder being determined automatically:
+        ```
+        extractor.save_to_folder(name="my_extractor_data")
+        ```
+        In this case, the data is saved in a subfolder named "my_extractor_data"
+        within the global temporary folder (set using `set_global_tmp_folder`). If no
+        global temporary folder is set, one will be generated automatically.
 
-        The folder must not exist. If it exists, remove it before.
+        3. If neither `name` nor `folder` is provided, a random name will be generated
+        for the subfolder within the global temporary folder.
 
         Parameters
         ----------
-        name: None str or Path
-            Name of the subfolder in get_global_tmp_folder()
-            If "name" is given, "folder" must be None.
-        folder: None str or Path
-            Name of the folder.
-            If "folder" is given, "name" must be None.
-        overwrite: bool, default: False
-            If True, the folder is removed if it already exists
+        name : str or Path, optional
```

Comment:
```suggestion
        name : str or Path, optional
```

currently this is wrong, because only a path would work.

And this type of optional is my pet peeve :)

Reply:
What do you mean? what are the preferences with option and what is your preference?

Reply:
Sorry so it is confusing, but there are two uses of optional--one from std library meaning `value | None`

```python
from typing import Optional
Optional[float]
```

And one from numpy docstring style, which is 

```python
def function(my_arg=1)
 """ my_arg: int, optional
```

notice the difference between Optional (from typing) and optional (lowercase numpy docstring convention)

The difference being Optional says an argument is actually optional (for example you choose one between folder or name so you have the option to pick). optional means that a default is already supplied to the argument of the function so you can optionally change the default value. 

I don't like the numpy style because I prefer to be more explicit in the docstring so instead I prefer
```
my_arg: int, default =1 
```

instead of, 
```
my_arg: int, optional
```

So I've worked toward removing all mentions of argument, default_value, optional from the library.
If someone prefers Optional[x] to mean a default of None that doesn't really bother me.

Reply:
technically Optional[float], optional would be an appropriate numpy docstring because a default is supplied in the arguments of the function. Which I don't like.

Reply:
I don't think that you are right about the meaning of `typing.Optional`:

https://docs.python.org/3/library/typing.html#typing.Optional

> Note that this is not the same concept as an optional argument, which is one that has a default. An optional argument with a default does not require the Optional qualifier on its type annotation just because it is optional. For example:

> def foo(arg: int = 0) -> None:
    ...
> On the other hand, if an explicit value of None is allowed, the use of Optional is appropriate, whether the argument is optional or not. For example:

Optional means that it can be None.

Reply:
But if anything that makes the need for your disctinction stronger,  if there is a default value, you should say what the default values is.

Reply:
I am getting you right if I think that the current docstring is working well with your preference?